### ISSUE TITLE
Better installation method

### ISSUE THREAD

**Issue by 00sapo** [arminstraub/krop#37]
A better and more portable installation method:

1. install [`pipx`](https://github.com/pypa/pipx#install-pipx)
2. install `krop`: `pipx install krop`
3. install dependencies: `pipx inject krop python-poppler-qt5 pypdf2`

**Comment by arminstraub** [comment]
Thank you for mentioning this!  (And apologies for the horribly late reaction...)  Did you have luck running krop on macOS that way?

**Comment by jeffkowalski** [comment]
Here is what I found to work on Linux Mint 22 Cinnamon

```
sudo apt install python3-poppler-qt5

pipx install git+https://github.com/arminstraub/krop
pipx inject krop PyQt5
pipx inject krop python-poppler-qt5
pipx inject krop pypdf

# install icons
{
    cd ~/.local/share/pipx/venvs/krop/share/icons/hicolor || exit
    find . -type f | while read -r file; do
        target_dir="$HOME/.local/share/icons/hicolor/$(dirname "$file")"
        mkdir -p "$target_dir"  # Ensure the target directory exists
        ln -sfn "$PWD/$file" "$target_dir/$(basename "$file")"
    done
    gtk-update-icon-cache ~/.local/share/icons/
}

# install desktop file
(
    cd ~/.local/share/pipx/venvs/krop/share/applications || exit
    find . -type f | while read -r file; do
        clean_path="${file#./}"  # Remove the leading "./"
        target="$HOME/.local/share/applications/$(basename "$clean_path")"
        ln -sfn "$PWD/$clean_path" "$target"
    done
    update-desktop-database ~/.local/share/applications/
)
```

While pipx installs the main entry, and the man page, the subsequent code installs the icons, and more importantly the desktop file that integrates with the system (right click menus, start/program menu, mime-type-aware file opens, etc.)

Note that under this installation method, we trigger certain libraries in krop, as seen when we invoke it:

```
$ krop
Using PopplerQt for rendering.
Using pypdf for cropping.
```

There are other renderers and croppers available.  Regrettably, I could not get any other potentially modern ones to be triggered properly.   I was unable to decipher the decision code here, which baffled me https://github.com/arminstraub/krop/blob/9fd6916b859c7fe5fae263dc5fc4f07c86134c13/krop/pdfcropper.py#L258

**Comment by arminstraub** [comment]
Thank you for the comments on how to install using ``pipx``!  As for using other croppers, what happens when you use something like the following?
```
$ krop --use-pikepdf
```
You should then see either "Using pikepdf for rendering" or "pikepdf was requested but failed to load" depending on whether pikepdf is installed on your system (and available to krop; if using pipx you probably have to inject it appropriately).  The code you mention first loads any cropper that was explicitly requested (using something like ``--use-pikepdf``) before trying them all in a certain order.

With the increased number of supported libraries, I have just updated the code for selecting which cropper to use (and hopefully, it is now easier to follow).  I couldn't test it thoroughly (it does work on my system) so please give it a try and let me know if something doesn't work!

**Comment by jeffkowalski** [comment]
As you expect, under the setup I described above, pikepdf doesn't load and the code reverts to another cropping tool.
```
λ krop --use-pikepdf
Using PopplerQt for rendering.
pikepdf was requested but failed to load.
Using pypdf for cropping.
```
I'll try to install pikepdf and re-test that in that scenario.

**Comment by jeffkowalski** [comment]
I started from the configuration steps I noted above, then injected pikepdf:
```
~ λ pipx inject krop pikepdf
  injected package pikepdf into venv krop
done! ✨ 🌟 ✨
```
The app starts fine, with the proper selection of cropping tool, as expected:
```
~ λ krop --use-pikepdf
Using PopplerQt for rendering.
Using pikepdf for cropping.
```
However, when given a file argument, it does not show the file contents (expected thumbnail images of the pdf)
```
~/tmp λ krop --use-pikepdf z.pdf 
Using PopplerQt for rendering.
Using pikepdf for cropping.
```
When I omit the use flag, cropping reverts back to pypdf, and I see the file contents as expected. 
```
~/tmp λ krop z.pdf 
Using PopplerQt for rendering.
Using pypdf for cropping.
```
I suspect there is some unfortunate interaction between selection/enabling of the rendering versus cropping service.

Hope this helps.

**Comment by arminstraub** [comment]
How strange!  I have not been able to reproduce these issues.  But I played around a bit with pipx and added some instructions to the krop website (including a link to your instructions for installing the desktop file).  In particular, I did not like injecting PyQt5 because, at least on my system, this resulted in krop looking different from other Qt applications (for instance, the fonts were smaller).  But using the flag ``--system-site-packages`` solved that issue for me because then there is no need to inject.  Also, have you tried injecting/using PyMuPDF?  That library can be used for both viewing and cropping.

BTW, I just released a new version (0.7.0).  Maybe that release solves some of your issues?

**Comment by jeffkowalski** [comment]
Yes, the new release is much better.  In particular the problem I was experiencing with ```--use-pikepdf``` not accepting the file on the commandline is now resolved.
I hear you about not wanting to inject PyQt5.  Perhaps PyQt6 is a better match with your system.  Of course, if you have either installed as a system site package, then the flag you suggest will pick that up.  I like using purer venvs because it helps narrow down problems if they occur, and I don't need to wonder which apps are using system packages that I may want to upgrade or remove.  
I confirm that krop is working well now either way - pure venv or using site packages.
Thank you for your great work!

**Comment by arminstraub** [comment]
@jeffkowalski Awesome!  Thank you for testing the new version!  I'm very happy to hear that it fixed the previous issue :)
